tests/kernel/wave/wave_e2e_test.py

```diff
@@ -17,11 +17,15 @@ def test_copy():
     N = tkl.sym.N
     BLOCK_M = tkl.sym.BLOCK_M
     BLOCK_N = tkl.sym.BLOCK_N
+    ELEMS_PER_THREAD = tkl.sym.ELEMS_PER_THREAD
     ADDRESS_SPACE = tkl.sym.ADDRESS_SPACE
 
     constraints: list[tkw.Constraint] = [
         tkw.HardwareConstraint(
-            threads_per_wave=64, waves_per_block=(1, 1, 1), vector_shapes={M: 16, N: 16}
+            threads_per_wave=64,
+            waves_per_block=(1, 1, 1),
+            # vector_shapes={M: 1, N: ELEMS_PER_THREAD}, # TODO: Symbol doesn't work here, but should
+            vector_shapes={M: 1, N: 4},

```

Comment:
I think 1, 2 and 8 would also work here because they map to the instructions. For 16, not sure what it was doing. We should probably add either an assert or break down larger loads to smaller ones.

Reply:
At least currently, this value must be exactly aligned with `elements_per_thread`, otherwise things will break. And in general `elements_per_thread` currently in some weird position, as never taken in account in expansion or any indexing calculations. Also, big vector sizes works too (I tried 16) as, apparently, underlying llvm pipeline is able to decompose them.